### PR TITLE
[ARTS-620] - Link Person to User with Authorisation

### PR DESCRIPTION
**Pull request by darrengsmithson**
## Description
Added authorisation code to link endpoint in PractitionerController. Added 'link-user' to permissions data.
An update to the init-service application.yml property files in mts-trial-deployment-config repo is needed to test this. API tests will fail unless 
a)docker-compose points at branch - https://github.com/NDPH-ARTS/mts-trial-deployment-config/tree/ARTS-620-auth-link-user - or 
b)we commit the config to main with..
https://github.com/NDPH-ARTS/mts-trial-deployment-config/pull/72

### Changes
- [ARTS-620] - Link Person to User with Authorisation

### Checklist:

- [ ] Branch name follows convention (feature/arts-#-short-name OR fix/arts-#-short-name)
- [ ] I have included a short-meaningful title, description and changes for this PR

[ARTS-620]: https://ndph-arts.atlassian.net/browse/ARTS-620

### CODE REVIEW

**Review by tamirkamara — APPROVED** [review]

**Review by LizaShak — COMMENTED** [review]
Just wanted to make sure -did you test it on azure? Was the init service able to link the role for the bootstrap user ?

File: role-service/src/main/resources/db/changelog/changelog-permission-data.xml
Comment (LizaShak) [inline role-service/src/main/resources/db/changelog/changelog-permission-data.xml:22]: I think it is better to use your own alias...

Comment (darrengsmithson) [reply role-service/src/main/resources/db/changelog/changelog-permission-data.xml:22]: ok will do..

**Review by LizaShak — APPROVED** [review]

**Review by tamirkamara — DISMISSED** [review]

**Review by LizaShak — APPROVED** [review]

**Comment by LizaShak** [comment]
Approved, please make sure once this PR is merged then at the mts-trial-deployment-config  we also have a PR to give that permission to the bootstrap user

File: docker-compose.yml
Comment (shajidissa) [inline docker-compose.yml:57]: @darrengsmithson this required ?

**Review by sameerapurini — APPROVED** [review]

**Review by shajidissa — APPROVED** [review]

**Pull request merged**
